src/zenml/logging/step_logging.py

```diff
@@ -76,188 +86,167 @@ def prepare_logs_uri(
     return logs_uri
 
 
-class StepStdOut(StringIO):
-    """A replacement for the sys.stdout to turn outputs into logging entries.
+class StepLogsStorage:
+    """Helper class which buffers and stores logs to a given URI."""
 
-    When used in combination with the ZenHandler, this class allows us to
-    capture any print statements or third party outputs as logs and store them
-    in the artifact store.
+    def __init__(
+        self,
+        logs_uri: str,
+        max_messages: int = STEP_LOGS_STORAGE_MAX_MESSAGES,
+        time_interval: int = STEP_LOGS_STORAGE_INTERVAL_SECONDS,
+    ) -> None:
+        """Initialization.
 
-    Right now, this is only used during the execution of a ZenML step.
-    """
+        Args:
+            logs_uri: the target URI to store the logs.
+            max_messages: the maximum number of messages to save in the buffer.
+            time_interval: the amount of seconds before the buffer gets saved
+                automatically.
+        """
+        # Parameters
+        self.logs_uri = logs_uri
+        self.max_messages = max_messages
+        self.time_interval = time_interval
 
-    stdout_logger = logging.getLogger(STEP_STDOUT_LOGGER_NAME)
+        # State
+        self.buffer: List[str] = []
+        self.disabled_buffer: List[str] = []
+        self.last_save_time = time.time()
+        self.disabled = False
 
-    def write(self, message: str) -> int:
-        """Write the incoming string as an info log entry.
+    def write(self, text: str) -> None:
+        """Main write method.
 
         Args:
-            message: the incoming message string
-
-        Returns:
-            the length of the message string
+            text: the incoming string.
         """
-        if message != "\n":
-            self.stdout_logger.info(message)
-        return len(message)
+        if text == "\n":
+            return
 
+        if not self.disabled:
+            self.buffer.append(text)
+        else:
+            self.disabled_buffer.append(text)
```

Comment:
I don't think you should use a separate buffer just to store messages that are generated by the process that saves logs to the artifact store. Those logs should basically never be stored in the artifact store. In other words, there should be no `else` here.

Reply:
removed the separate buffer.